modules/seqexec/server/src/main/scala/seqexec/server/SeqTranslate.scala

```diff
@@ -167,16 +167,17 @@ class SeqTranslate(site: Site, systems: Systems[IO], settings: TranslateSettings
 
   }
 
-  private def handleError(t: TrySeq[Unit]): executeEngine.EventType = t match {
-    case Left(e) => Event.logErrorMsg(SeqexecFailure.explain(e))
+  private def handleError: Either[Throwable, Unit] => executeEngine.EventType = {
+    case Left(e: SeqexecFailure) => Event.logErrorMsg(SeqexecFailure.explain(e))
+    case Left(e: Throwable) => Event.logErrorMsg(SeqexecFailure.explain(SeqexecFailure.SeqexecException(e)))
     case _       => Event.nullEvent
   }
 
   def stopObserve(seqId: Observation.Id)(
     implicit cio: Concurrent[IO],
              tio: Timer[IO]
   ): EngineState => Option[Stream[IO, executeEngine.EventType]] = st =>{
-    def f(oc: ObserveControl[IO]): Option[SeqAction[Unit]] = oc match {
+    def f(oc: ObserveControl[IO]): Option[IO[Unit]] = oc match {
```

Comment:
This is a weird type. Seems like you want `IO[Unit]` here where the empty case is `IO()`.

Reply:
I'll review this. In cases we need to have `Option[IO[Unit]]` as we need to know if the `IO` does nothing. OTOH I suppose there is no harm in jut running `IO.unit`